Dockerfile

```diff
@@ -1,17 +1,29 @@
 FROM chainguard/wolfi-base:latest
 
-# Set the working directory
+# Add non-root user
+USER nonroot
+
+# Create necessary directories with correct permissions
 WORKDIR /inbox451
 
-# Copy only the necessary files
-COPY inbox451 .
-COPY config.yml.sample config.yml
+# Copy binary
+COPY --chown=nonroot:nonroot inbox451 .
 
-# Expose the application ports
-# 8080: HTTP
-# 1025: SMTP
-# 1143: IMAP
+# 8080: HTTP API
+# 1025: SMTP Server
+# 1143: IMAP Server
 EXPOSE 8080 1025 1143
 
-# Define the command to run the application
-CMD ["./inbox451"]
+# Health check
+HEALTHCHECK --interval=30s --timeout=3s --start-period=5s --retries=3 \
+  CMD wget --no-verbose --tries=1 --spider http://localhost:8080/api/health || exit 1
```

Comment:
@Jalmeida1994 should we interpolate the HTTP port here instead of hardcoding the 8080 port